### DISCUSSION TITLE
BEX communication limitation/issues?

### DISCUSSION THREAD

**Discussion by pandafox** [quasarframework/quasar#15318] [CLI - BEX mode]
Goal sending event from app to content/dom and get result of execution back
From the doc we need to chain/relay the event calls w background ex: app > background > content/dom
https://quasar.dev/quasar-cli-vite/developing-browser-extensions/bex-communication#communication-rules
> If you want to allow your Dev Tools, Popup or Options Page BEX to communicate with a web page, you will need to use the background script as a proxy. You would do this by:

> Adding a listener on the background script which in turn emits another event.
Add a listener to your Quasar App running in the Web Page context which listens for the event the background script is raising
Emitting the event to your background script from your Dev Tools, Popup or Options Page.

> So far we have tested the following scenarios that works/doesn't work, but we aren't really sure why.

Works
1. app > background - can use await to obtain the response
2. background > content - can use await to obtain the response

Doesn't work
1. background > dom - message ignored/goes into void, await never returns
2. app > background > content using await (respond is called in content, and the result passed back by calling respond in background) - stuck with background await call, content received and respond called but never returned back to background

**Comment by Flamenco** [comment]
I have same issue:  Responding to the background script does not work.

background
```ts
bridge.send('echo', 'hello').then(it=>{
  // THIS IS NEVER CALLED
  console.log(it.data)
})
```

content
```ts
bridge.on('echo', ({data, respond}) => {
  respond(data)
})
```
